frontend/connection/src/DoInitPings.tsx

```diff
@@ -153,9 +204,38 @@ If you are trying to access a Streamlit app running on another server, this coul
           // The request was made but no response was received
           // `error.request` is an instance of XMLHttpRequest in the browser and an instance of
           // http.ClientRequest in node.js
+
+          if (shouldSendClientError) {
+            // Handle retrieving the source URL from the error (health or host-config endpoint)
+            const source = error.request.path
+              ? new URL(error.request.path).pathname
```

Comment:
Just checking, is there any way `error.request.path` could fail to be parsed as a URL? Would wrapping in a `try-catch` be a little safer?